django/db/backends/postgresql/features.py

```diff
@@ -67,3 +67,24 @@ def is_postgresql_10(self):
     has_brin_autosummarize = property(operator.attrgetter('is_postgresql_10'))
     has_phraseto_tsquery = property(operator.attrgetter('is_postgresql_9_6'))
     supports_table_partitions = property(operator.attrgetter('is_postgresql_10'))
+
+    # Views do not support the allows_group_by_selected_pks optimization"
+    def allows_group_by_selected_pks_on_model(self, model):
+        if model._meta.managed:  # We are assuming managed tables are not views, since Django can't create views.
+            return True
+
+        try:
+            return not model._meta._is_view
+        except AttributeError:
+            with self.connection.cursor() as cursor:
+                cursor.execute("""
+                    SELECT 1
+                    FROM pg_class c
+                        JOIN pg_namespace n ON c.relnamespace = n.oid
+                    WHERE c.relname = %s
+                        AND c.relkind = 'r'
+                        AND n.nspname NOT IN ('pg_catalog', 'pg_toast')
+                        AND pg_catalog.pg_table_is_visible(c.oid)
+                     """, (model._meta.db_table,))
```

Comment:
I think we should move this hook to the PostgreSQL's introspection e.g. `is_table(table_name)`:
```python
def is_table(self, table_name):
    with self.connection.cursor() as cursor:
        cursor.execute("""
            SELECT TRUE
            FROM pg_class c
            JOIN pg_namespace n ON c.relnamespace = n.oid
            WHERE
               c.relname = %s AND
               c.relkind = 'r' AND
               n.nspname NOT IN ('pg_catalog', 'pg_toast') AND
               pg_catalog.pg_table_is_visible(c.oid)
       """, [model._meta.db_table])
       return cursor.fetchone() is not None
```
Also this should be cached e.g. `lru_cache(maxsize=512)` we don't want to perform an extra query each time.